src/main/java/com/google/api/codegen/transformer/java/JavaSurfaceTransformer.java

```diff
@@ -976,8 +1049,51 @@ private SettingsDocView generateSettingsDoc(
     return settingsDoc.build();
   }
 
+  private ApiAndSamples<StaticLangApiMethodView, StaticLangApiMethodView> methodAndSamplesFor(
+      SurfaceNamer namer, StaticLangApiMethodView methodView) {
+
+    Set<SampleValueSetView> matchingValueSets =
+        methodView.sampleValueSetsCollection().forSampleType(SampleType.STANDALONE);
+
+    SampleInfo.Builder<StaticLangApiMethodView> sampleInfoBuilder =
+        SampleInfo.<StaticLangApiMethodView>newBuilder();
+
+    List<SampleInfo<StaticLangApiMethodView>> sampleInfos = new ArrayList<>();
+    for (SampleValueSetView values : matchingValueSets) {
+      sampleInfos.add(
+          sampleInfoBuilder
+              .valueSet(values)
+              .className(
+                  namer.getApiSampleClassName(
+                      methodView.name(),
+                      // TODO(vchudnov-g): Simplify so we don't need all these casing conversions
+                      Name.anyCamel(methodView.type().toString()).toLowerCamel(),
```

Comment:
Avoid casing conversions in transformer code

Reply:
Yeah, not happy about that; that's what I put the TODO. I'll address it earlier, though, if you think it's a blocker.

Reply:
Done